src/main/kotlin/uk/gov/justice/digital/hmpps/mailboxregisterapi/mailboxes/offendermanagementunits/OffenderManagementUnitMailboxService.kt

```diff
@@ -31,11 +34,24 @@ class OffenderManagementUnitMailboxService(
     return repository.saveAndFlush(existingMailbox)
   }
 
-  @Transactional
-  fun listMailboxes(): List<OffenderManagementUnitMailbox> = repository.findAll(Sort.by(Sort.Direction.ASC, OffenderManagementUnitMailbox::createdAt.name))
+  fun listMailboxes(
+    prisonCode: PrisonCode?,
+    role: OffenderManagementUnitRole?,
+  ): List<OffenderManagementUnitMailbox> = repository.findAll(
+    Example.of(OffenderManagementUnitMailbox(role = role, prisonCode = prisonCode), FILTER),
+    SORTED_BY_CREATED_AT_ASC,
+  )
 
   @Transactional
   fun deleteMailbox(id: UUID) {
     repository.delete(getMailboxById(id))
   }
+
+  companion object {
+    val SORTED_BY_CREATED_AT_ASC: Sort = Sort.by(Sort.Direction.ASC, OffenderManagementUnitMailbox::createdAt.name)
+    val FILTER: ExampleMatcher = ExampleMatcher.matchingAll()
+      .withMatcher("role", ExampleMatcher.GenericPropertyMatchers.exact())
+      .withMatcher("prisonCode", ExampleMatcher.GenericPropertyMatchers.exact())
```

Comment:
Considering this is doing exact matching on two fields there's no difference between adding a [spring data query builder](https://docs.spring.io/spring-data/jpa/reference/repositories/query-methods-details.html#repositories.query-methods.query-creation) method to generate the impl for you?

Reply:
@andrewrlee Can you give me some pointers

I tried the following doesn't give the same behaviour:
```
fun findByRoleOrPrisonCode(role: OffenderManagementUnitRole?, prisonCode: PrisonCode?, sort: Sort?): List<OffenderManagementUnitMailbox>
```


```
should return a list of existing mailboxes with an optional filter by role AND prison code

Expected size: 1 but was: 3 in:
```

Reply:
It needs to be

```
fun findByRoleOrPrisonCodeOrBothIfPresent(... optional fields ...)
```